lib/Runtime/Types/PathTypeHandler.cpp

```diff
@@ -223,8 +223,7 @@ namespace Js
 
         // Check numeric propertyId only if objectArray available
         uint32 indexVal;
-        ScriptContext* scriptContext = instance->GetScriptContext();
-        if (instance->HasObjectArray() && scriptContext->IsNumericPropertyId(propertyId, &indexVal))
+        if (instance->HasObjectArray() && requestContext->IsNumericPropertyId(propertyId, &indexVal))
```

Comment:
Why are we using requestContext here?

Reply:
Good question. Why we shouldn't? It will look through ThreadContext either way. No?

Reply:
Codeflow seemed to lose my comment, apologies if it's repeated..

In this particular case it doesn't matter. The general convention throughout the codebase is to only use requestContext if you have a specific CrossSite marshalling concern.

In a stable class like this we wouldn't expect this to change except in rare circumstances.

As a thought exercise, if these would produce different results I would still expect the instance's scriptContext to be correct since you are asking the object whether this is a numeric id for it's purposes.

Reply:
>  I would still expect the instance's scriptContext to be correct since you are asking the object whether this is a numeric id for it's purposes.

Makes sense. Will roll back the change.